### PR TITLE
New zigpy init/01 new zigpy init

### PR DESCRIPTION
**Pull request by Adminiuga**
Implement new zigpy initialization and support for radio lib configuration schemas in https://github.com/zigpy/zigpy/commits/new-zigpy-init/dev

This would allow each radio lib to extend the zigpy configuration schema by adding radio specific configuration options.
Corresponding HA changes are currently in https://github.com/Adminiuga/home-assistant/tree/experimental/new-zigpy-init

I don't have a zigate so can't test, but once tested I'm going to publish new zigpy version.

### CODE REVIEW

**Comment by Hedda** [comment]
@doudz do you think that could extend on this to configure pizigate and/or WiFi ZiGate, like specify IP address and port, example : socket://192.168.1.10:9999 and other custom configs for the ZiGate radios?

**Comment by Adminiuga** [comment]
The build is failing because of the latest change in Zigpy requiring ApplicationController to have a `probe()` method. Needs #28 as well. 

https://github.com/zigpy/zigpy-zigate/tree/new-zigpy-init/dev has both in and passes the build.

**Comment by Adminiuga** [comment]
>like specify IP address and port, example : socket://192.168.1.10:9999 and other custom configs for the ZiGate radio

@Hedda have you tried it? I have a feeling it should work out of the box, depending on the pyserial-asyncio library. At least from the looks of it.

File: zigpy_zigate/zigbee/application.py [hunk 1]
```diff
@@ -26,6 +31,8 @@ def __init__(self, api, database_file=None):
 
     async def startup(self, auto_form=False):
         """Perform a complete application startup"""
+        self._api = ZiGate.new(self._config[CONF_DEVICE], self)
```
Comment (doudz) [inline zigpy_zigate/zigbee/application.py:34]: await is missing
should be `self._api = await ZiGate.new(self._config[CONF_DEVICE], self)`

**Comment by doudz** [comment]
> > like specify IP address and port, example : socket://192.168.1.10:9999 and other custom configs for the ZiGate radio
> 
> @Hedda have you tried it? I have a feeling it should work out of the box, depending on the pyserial-asyncio library. At least from the looks of it.

This part doesn't really work because for some reasons, socket doesn't work as expected with pyserial.
I think we have to use asyncio socket

**Pull request closed without merging**